libs/ledger-live-common/src/families/cosmos/js-updateTransaction.ts

```diff
@@ -1,15 +1,18 @@
 import { Transaction } from "./types";
+import { defaultUpdateTransaction } from "@ledgerhq/coin-framework/bridge/jsHelpers";
 
-const updateTransaction = (t: Transaction, patch: Partial<Transaction>): Transaction => {
-  if ("mode" in patch && patch.mode !== t.mode) {
-    return { ...t, ...patch, gas: null, fees: null };
-  }
+const updateTransaction = (tx: Transaction, patch: Partial<Transaction>): Transaction => {
+  const updatedTx = defaultUpdateTransaction(tx, patch);
 
-  if ("validators" in patch && patch.validators?.length !== t.validators.length) {
-    return { ...t, ...patch, gas: null, fees: null };
+  if (
+    ("mode" in patch && patch.mode !== tx.mode) ||
+    ("validators" in patch && patch.validators?.length !== tx.validators.length)
+  ) {
+    updatedTx.gas = null;
+    updatedTx.fees = null;
   }
 
-  return { ...t, ...patch };
+  return updatedTx;
 };
```

Comment:
I think this can lead to unexpected mutation of the inital `tx`. Maybe we should rather create a new `patch` instead and use a spread still have `defaultUpdateTransaction` as the last instruction with the 2 patches ?

Reply:
You're right there an issue here, but I think it's actually the opposite :smile: 

What matters is that _reference_ doesn't change when data doesn't change, which is actually different from _immutability_.

```
const a = { foo: 1 };
const b = a;

a.foo = 2; // mutate
console.log(a === b); // true

const c = { ...a, foo: 3 }; // change ref
console.log(a === c); // false
```

However the issue currently, is that the code in the `if` can actually mutate the tx _without_ changing its ref, which is also not what we want.

Reply:
Yup that's what I meant, in the case where `defaultUpdateTransaction` sees no changes from the patch, it will return `tx` (and its ref) and that means that you're going to update the content of `tx` (now `updatedTx`) without changing the ref 💥 🧠

Reply:
Haha ok didn't get that's what you meant.
This should be fixes in the last commits.